### PR TITLE
File Connector 

### PR DESCRIPTION
**Pull request by yashothara**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/FileSend.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.Connector;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+import org.wso2.carbon.connector.util.FileConnectorUtils;
+import org.wso2.carbon.connector.util.FileConstants;
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+
+public class FileSend extends AbstractConnector implements Connector {
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/FileSend.java]: Add class level comment.

File: src/main/java/org/wso2/carbon/connector/FileSend.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        } catch (XMLStreamException e) {
+            handleException(e.getMessage(), e, messageContext);
+        } catch (IOException e) {
+            handleException(e.getMessage(), e, messageContext);
+        } catch (JSONException e) {
+            handleException(e.getMessage(), e, messageContext);
+        }
+    }
+
+    private MessageFormatter getMessageFormatter(org.apache.axis2.context.MessageContext msgContext) {
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/FileSend.java]: Add method level comment.

File: src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        // SSH Key checking
+        SftpFileSystemConfigBuilder.getInstance().setStrictHostKeyChecking(opts, "no");
+
+        // Root directory set to user home
+        SftpFileSystemConfigBuilder.getInstance().setUserDirIsRoot(opts, false);
+
+        // Timeout is count by Milliseconds
+
+        SftpFileSystemConfigBuilder.getInstance().setTimeout(opts, 100000);
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java]: Remove the extra spaces.

File: src/main/java/org/wso2/carbon/connector/FileMove.java [hunk 4]
```diff
@@ -43,7 +44,6 @@ public void connect(MessageContext messageContext) {
                 FileConstants.FILE_LOCATION);
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/FileMove.java]: Format the code

File: src/main/java/org/wso2/carbon/connector/FileMove.java [hunk 5]
```diff
@@ -35,6 +32,10 @@
 import org.wso2.carbon.connector.util.FileConstants;
 import org.wso2.carbon.connector.util.ResultPayloadCreate;
 
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.IOException;
+
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/FileMove.java]: Please add the class level comments.

File: src/main/java/org/wso2/carbon/connector/FileSearch.java [hunk 6]
```diff
@@ -39,6 +36,9 @@
 import org.wso2.carbon.connector.util.FileConstants;
 import org.wso2.carbon.connector.util.FilePattenMatcher;
 import org.wso2.carbon.connector.util.ResultPayloadCreate;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/FileSearch.java]: Please add class level comments

File: src/main/java/org/wso2/carbon/connector/FileSend.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+
+/**
+ * Send the file to specific location.
+ */
+public class FileSend extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(FileSend.class);
+
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/FileSend.java]: Add method level comments. This should be fixed throughout PR.

File: src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemOptions;
+import org.apache.commons.vfs2.provider.ftp.FtpFileSystemConfigBuilder;
+import org.apache.commons.vfs2.provider.ftps.FtpsFileSystemConfigBuilder;
+import org.apache.commons.vfs2.provider.sftp.SftpFileSystemConfigBuilder;
+
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java]: Add class level comments

File: src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        SftpFileSystemConfigBuilder.getInstance().setTimeout(opts, 100000);
+
+        FtpFileSystemConfigBuilder.getInstance().setPassiveMode(opts, true);
+
+        FtpFileSystemConfigBuilder.getInstance().setSoTimeout(opts, 100000);
+
+        FtpsFileSystemConfigBuilder.getInstance().setPassiveMode(opts, true);
+
+        return opts;
+
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java]: Remove empty space and do the code formatting.

File: src/main/java/org/wso2/carbon/connector/util/FileConstants.java [hunk 10]
```diff
@@ -56,4 +56,9 @@
     public static final String TARGET_FILE= "targetFile";
     public static final String BINARY_TRANSFER= "binaryTransfer";
     public static final String LOCAL_ACTIVE= "localActive";
+    public static final String STREAMING= "streaming";
+    public static final String ADDRESS= "address";
+    public static final String APPEND= "append";
+    public static final String DEFAULT_RESPONSE_FILE = "/response.xml";
+
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/util/FileConstants.java]: Remove empty line

File: src/test/resources/artifacts/ESB/connector/config/fileconnector.properties [hunk 11]
```diff
@@ -29,6 +31,10 @@ binaryTransfer=""
 localActive="""
 keepAliveTimeout=""
 controlKeepAliveReplyTimeout=""
+streaming=true
+
+address=file:///home/yasho/Desktop/testESB/OutTest
+append=true
```
Comment (Kanapriya) [inline src/test/resources/artifacts/ESB/connector/config/fileconnector.properties]: Remove unwanted lines

File: src/main/java/org/wso2/carbon/connector/FileMove.java (hunk 4 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/FileMove.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/FileMove.java (hunk 5 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/FileMove.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/FileSearch.java (hunk 6 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/FileSearch.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/FileSend.java (hunk 7 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/FileSend.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java (hunk 8 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java (hunk 9 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/util/FTPSiteUtils.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/main/java/org/wso2/carbon/connector/util/FileConstants.java (hunk 10 above)
Comment (yashothara) [inline src/main/java/org/wso2/carbon/connector/util/FileConstants.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/test/java/org/wso2/carbon/connector/integrationTest/FileConnector/FileConnectorIntegrationTest.java [hunk 12]
```diff
@@ -357,4 +355,31 @@ public void testFtpOverProxyWithNegativeCase() throws Exception {
...
+    @Test(groups = {"wso2.esb"}, description = "FileConnector VFS Send file integration test with " +
+            "Negative parameters")
+    public void testSendFileWithNegativeCase() throws Exception {
+        esbRequestHeadersMap.put("Action", "urn:send");
+        RestResponse<JSONObject> esbRestResponse =
+                sendJsonRestRequest(proxyUrl, "POST", esbRequestHeadersMap,
+                        "FileSendMandatoryNegative.json");
+        Assert.assertEquals(esbRestResponse.getHttpStatusCode(), 202);
+    }
+
```
Comment (yashothara) [inline src/test/java/org/wso2/carbon/connector/integrationTest/FileConnector/FileConnectorIntegrationTest.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

File: src/test/resources/artifacts/ESB/connector/config/fileconnector.properties (hunk 11 above)
Comment (yashothara) [inline src/test/resources/artifacts/ESB/connector/config/fileconnector.properties]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/19.

**Pull request merged**